### PR TITLE
[Actually merged, ignore Git] Mycogenesis and Mycus Provenance mutations

### PR DESCRIPTION
**Pull request by ghost**
It's another Mycus patch! The short version is this.
* Added the Mycogenesis mutation, which is the final form of Mycus Fireproofing. That weak human flesh has been fully spliced with Mycus physiology; the end result is that, not only does it offer slightly more armor, but you may easily sleep on any fungal tile, and doing so will place you into a semi-dormant state, conferring the following effects:
  * You require much less sleep (-5 fatigue every 10 minutes in addition to normal sleep) than if you'd slept on somewhere that isn't a fungal tile.
  * You are less affected by strong temperatures - mechanically identical to the Climate Control CBM.
  * You generate a cloud of spores around you every hour, similar to a fungaloid.
* Added the Mycus Provenance mutation, which does away with the silliness of Mycus Bloom and Mycus Spores flipflopping constantly in sleep. Provenance combines the two; using it will both create a haze cloud like Blossoms, and create several spores like Sporogenesis. This mutation can be gained through either of its prerequisites, and will not regress back to either of them.
* Adjusted descriptions and messages for a bunch of Mycus mutations to clarify their function.
* Adds a bunch of new survivor note possibilities themed around the Mycus.
* Mycus Feeder will keep hunger and thirst synchronized for the player, so that their only source (mycus fruits) don't become suddenly unreliable in sating both needs.

Thematically, Mycogenesis serves to represent the completeness of adaptation to the Mycus - enough that you serve as a fungaloid while asleep - and Mycus Provenance serves to combine the flavor of both the fungal blossom and fungal flowers overmaps, as you now use the method that both of them use to spread the influence of the Mycus. With the flesh of the tower, the spores of the spire, and the haze of the blossom, you are the summary of the Mycus' influence on Earth, packaged into one local guide.

Mechanically, Mycogenesis serves as another incentive to spread the Mycus, at least a little - it's easier to sleep there. Provenance allows a Mycus character to gain the benefits of both options of spreading the Mycus influence without having to worry about having them randomly swapped - both converting zombies into fungal variants (Sporogenesis) and converting terrain into fungal variants (Blossoms), which is what the Mycus would prefer for its guides to begin with.

Mycus rations are intended to give Mycus characters a slight bit more freedom when delving into things such as science labs. Normal characters have the ability to bring foods like pemmican and granola (that are very light) and water (which is water), but mycus fruits are quite heavy and so bringing more than just a snack can often set back the amount you can lug out by quite a lot. Mycus rations serve to offset that.

[Relevant image](https://cdn.discordapp.com/attachments/252370797337772050/473585582853980171/mycusgf.png).

### CODE REVIEW

**Comment by nexusmrsep** [comment]
Please consider adding some, if not all, of those texts in the image to survivor notes in snippets.json to enhance the Mycus lore in the game.

**Comment by Night-Pryanik** [comment]
I like the changes, except for require much less sleep on fungal ground. What's the rationale for this? I suppose it would be better to just make fungal ground much more attractive and comfortable to sleep on, maybe even better than on beds.

**Comment by Night-Pryanik** [comment]
> can draw strength from them

Strength - yes, but I think this can't applied to reduction of fatigue and recovering from tiredness.

**Comment by Xhuis** [comment]
I disagree; sleep is the process of recovering your energy and preparing your body for the next day. I don't see why such an alien connectivity couldn't help you do so, especially because the Mycus already possesses the power to mutate you in your sleep *without* a direct connection to it.

**Comment by Xhuis** [comment]
I added a bunch of survivor note possibilities based on fungus. Loads of fun to write, too

**Comment by nexusmrsep** [comment]
They look better then I anticipate. Beyond creepy. Cudos to you. Great job.

File: data/json/snippets.json [hunk 1]
```diff
@@ -1434,6 +1434,86 @@
...
+  },
+  {
+    "type": "snippet",
+    "category": "note",
+    "text": [ "\"red blue yellow berry seed sap bloom flowers tower mycus mycus mycus.\"" ]
+  },
+  {
+    "type": "snippet",
+    "category": "note",
+    "text": [ "\"Husband 30 yrs died in bombs.  Mushrooms brought him back.  GIVE YOURSELF TO THEm AND THEY WILL HELP YOU\"" ]
```
Comment (Night-Pryanik) [inline data/json/snippets.json]: THEm -> THEM

Comment (Xhuis) [reply data/json/snippets.json]: Intentional misspelling; it's written by hand in-game, so little errors like that are fair play.

**Comment by Mecares** [comment]
Quench from dehydrated fruit makes no sense. Reducing weight by 80% using dehydration  while maintaining 63% of quench makes no sense.
Description would be better if it said "long journeys" instead of labs.

**Comment by Xhuis** [comment]
Mechanically speaking, removing the quench value entirely doesn't work at *all*, because Mycus characters can only eat mycus fruits (and with this PR, mycus rations.) There would be no point to bringing the rations if they didn't nourish you in both ways.

I decided to describe them as dehydrated and such in that the "quench" value doesn't come from actual hydration, but from the nutrition value of the fruits themselves. It isn't realistic that a fruit would quench more than an entire 500 mL bottle of water, no matter how alien it is, and the Mycus Feeder mutation states that nutritional requirements have been streamlined to the Mycus. As such, it's a reasonable guess that the fruits retain their usefulness in both ways even when dehydrated.

The description does not cite labs. That's a json comment.

**Comment by Mecares** [comment]
Dehydration and rehydration does not require magic mumbo jumbo, or at least less  which makes it the preferred solution here.

**Comment by Xhuis** [comment]
This is a pull request primarily addressing the specifics of being a person who is being slowly converted into an alien mushroom/human hybrid by a sentient fungal hivemind that invaded Earth when portals opened up to another dimension. I think "magic mumbo jumbo" is hardly a valid complaint at this point, and I feel that the reasons I outlined in the post above are more than enough to justify it.

**Comment by DracoGriffin** [comment]
There is some biological basis you can wrap your head around using the [Kangaroo rat](https://en.wikipedia.org/wiki/Kangaroo_rat#Range_and_habitat). In consideration of Mycus alienness and capability, we can _kinda_ establish they are a bit more aware and capable of this (whether by design of the fruit or efficiency of Mycus metabolism or both).

>They obtain enough water from the metabolic oxidation of the seeds they eat to survive and do not need to drink water at all.

So even though dehydrated, there is still chemical reaction capable of producing the hydration necessary but aa severely reduced costs (quench would need to be reduced a bit further, I think).

**Comment by nexusmrsep** [comment]
In other approach, if its fine by lore, Mycus might not need to drink at all.

**Comment by Mecares** [comment]
If you think mycus do not need water unlike all known real fungi give them low thirst into their mutation path that is what that mutation is in game  for  but quench and thirst are representing literally physically  water. But magic water for convenience against realism is not the preferable solution if a more plausible is available.

**Comment by Xhuis** [comment]
The Mycus is an alien organism. Its physiology doesn't follow the conventional boundaries of what we consider normal, given that inhalation of a single spore allows them to germinate inside of your bloodstream and burst through your skin if they cultivate enough. The blob (y'know, the thing that makes the zombies) doesn't require food at all, and triffids are implied to only consume the Mycus for their food source, given the existence of triffids specially evolved for hunting them.

You seem to not be taking my earlier points into account, so to avoid talking in circles, I'm going to call it a night here and head to bed. I appreciate the concern, but right now my tl;dr view is that, as a creature foreign to Earth, they don't follow Earth rules and require something outside of those rules for its sustenance. In fact, post-thresh Mycus characters physically *cannot* consume normal water; they have to consume the fruit for both satiation and quench, further reinforcing my idea that it's the sole source of what they need to survive.

**Comment by nexusmrsep** [comment]
What @Mecares is saying here is that youahve to answer the question about alien biology first - does Mycus needs water to live. If yes, then how would it aquire it? Becouse logic says dehydrated food is dehydrated, equals no water. If no however, then the need for water can be suppressed in the code, and mycus fruit can be dehydrated without adding quench values to it.

**Comment by Mecares** [comment]
You can't have Mucus only follow the rules when you like it. You can't have the fruit contain water when you want to dehydrate it for more convenience but say it does not contain water when it suits you for your alien do not need water to drink. Argument. I do not care about being exact like reality realism concerning extra dimensial alien life but you need to be plausible and more than anything else consistent.

**Comment by sfsworms** [comment]
Well, in any case, would it be that bad to add a recipe to re-hydrate the mycus ration and restore quench? Maybe Mycus can't drink water, but can absorb it when it's bound to whatever is in the mycus fruit.

**Comment by Lorith** [comment]
Something to keep in mind here is that Mycus have [no options at all](https://cdn.discordapp.com/attachments/473589967491170309/473596151849943060/unknown.png) except mycus fruit.  They cannot eat anything else, and they cannot drink anything at all.  You have to keep the hunger and thirst aligned or else you will be stuck with hunger or thirst and no way to deal with it except trying to arrange things such that you manage to get sick and puke, allowing you to eat more.  This issue is made far worse if you happen to have any of the hunger/thirst altering mutations or cbms, leading to a constant cycle.  If your hunger is full, you can't do anything for thirst.  If your thirst is full, you can't do anything for hunger.

Personally I am not sure on having compressed food, it seems fine to me that you would have to leave and make multiple trips, as it is a relatively small matter that can be solved with proper planning ahead of time.  But there does need to be a way to deal with hunger/thirst that are not aligned with each other.

The Mycus Provenance mutation is something I've been wanting for a long time, hope to see it soon.  I'm tired of having to try to rush everyone into power armor or cleansuits to avoid accidentally killing them in my sleep because my controllable mutation randomly flip-flopped to uncontrollable.

**Comment by nexusmrsep** [comment]
@Lorith in that case I'd personly opt for removing the need for water in general (why would you have ned for something you can't drink) and remove water from the dehydrated mycus fruits and it would all perfectly fit together.

Another option would be introducing something  along the lines of the plant mutation treshold where roots can be planted in soil to gain food and water - in that case sleeping on fungal bed would give you needed hydration. Let mycus quench the mycus I say. Something to consider.

**Comment by Lorith** [comment]
Personally I'm thinking something like fruit juice - squeeze it and get juice and pulp, then add in things you can do with them besides consume them, such as maybe deployable fungal wall/decoy things, or melee defense units or traps.  Lots of different options that could be done.

**Comment by Xhuis** [comment]
I've scrapped mycus rations, as they seem to be a great deal of trouble more than they're worth, but mainly for the second update in this commit. I'm not very willing to remove their quenching factor because of the fact that they can only get food and water from one source, which leads to said change in this commit!

As a Mycus Feeder, hunger and thirst are now synchronized. That being, if your hunger is higher than your thirst, your thirst will be set to your hunger, and vice-versa. In an ideal state, this mechanic won't be visible to the player, since mycus fruits have 80 quench and 80 sate, but as external factors may skew the rates at no fault of the player, this should help to align them.

**Comment by kevingranade** [comment]
Just dont call the rations dehydrated and it should be fine.

I need some clarity around the direction you're taking this, because it kind of seems like you're systematically eliminating the weaknesses of being mycus adapted.

The core limitation is that at the upper ends of mycus adaptation, the player is intended to be dependent on mycus structures for everything, food, water, rest, etc.  Avoid features that allow the player independence from mycus, they need just enough to go out and spread mycus more, and that's all.

**Comment by Xhuis** [comment]
My current vision for the Mycus is to increase their strength in their ability to spread the Mycus, as well as to solidify the mutations built around doing so. For instance, right now, with how the two "spreading" mutations flipflop, converting terrain is difficult because it's not reliable. With Provenance, they're able to quickly spread it much more reliably than before; additionally, its ability to create fungal blossoms and fungaloids makes it better for doing so.

The only benefit this PR confers with regards to independence is slightly increased armor with Mycogenesis, unless I'm mistaken. Requiring less sleep at fungal areas rewards the player for spreading the Mycus, and Provenance allows them to spread it well or seed an area with blossom or fungaloids to ensure that infestation is absolute there.

I *do* want to still *allow* Mycus characters to play the core game without shifting 100% of their focus to its spread, just the majority of it. Requiring the Mycus for food, water, and rest is all fine and dandy, but if we limit it too much then taking the path becomes a slog of just poofing, then sleeping, as your ability to explore and loot would be severely hampered by your inability to stray too far.

**Comment by Xhuis** [comment]
Does anything else need doing?

**Comment by kevingranade** [comment]
IT was a messy merge so github didn't pick up on it, but it's in.

**Comment by Xhuis** [comment]
Oh, good deal. I saw the red icon and was all like `:c`.

**Pull request closed without merging**